components/brave_ads/core/internal/history/ad_history_manager.cc

```diff
@@ -158,33 +169,34 @@ mojom::UserReactionType AdHistoryManager::DislikeAd(
     NotifyDidDislikeAd(mutable_ad_history_item);
   }
 
-  return mutable_ad_history_item.ad_user_reaction_type;
+  std::move(callback).Run(/*success=*/true);
 }
 
-mojom::UserReactionType AdHistoryManager::LikeCategory(
-    const AdHistoryItemInfo& ad_history_item) const {
+void AdHistoryManager::LikeCategory(const AdHistoryItemInfo& ad_history_item,
+                                    ToggleUserReactionCallback callback) const {
   const mojom::UserReactionType toggled_user_reaction_type =
       ClientStateManager::GetInstance().ToggleLikeCategory(ad_history_item);
   if (toggled_user_reaction_type == mojom::UserReactionType::kLike) {
     NotifyDidLikeCategory(ad_history_item);
   }
 
-  return toggled_user_reaction_type;
+  std::move(callback).Run(/*success=*/true);
 }
 
-mojom::UserReactionType AdHistoryManager::DislikeCategory(
-    const AdHistoryItemInfo& ad_history_item) const {
+void AdHistoryManager::DislikeCategory(
+    const AdHistoryItemInfo& ad_history_item,
+    ToggleUserReactionCallback callback) const {
   const mojom::UserReactionType toggled_user_reaction_type =
       ClientStateManager::GetInstance().ToggleDislikeCategory(ad_history_item);
   if (toggled_user_reaction_type == mojom::UserReactionType::kDislike) {
     NotifyDidDislikeCategory(ad_history_item);
   }
 
-  return toggled_user_reaction_type;
+  std::move(callback).Run(/*success=*/true);
```

Comment:
This is a prerequisite for future work to transition from JSON to a database. That work will then return either true or false, instead of always true.